src/components/AppSidebar/AppSidebar.vue

```diff
@@ -314,6 +329,19 @@ export default {
 			this.$emit('update:starred', this.isStarred)
 		},
 
+		editTitle(event) {
+			// Don't edit the title if editing is disabled
+			if (this.titleEditable === false) {
+				return
+			}
+			this.editing = true
+			// Focus the title input
+			this.$nextTick(
+				() => this.$refs.titleInput.focus()
+			)
+			this.$emit('start-editing')
+		},
```

Comment:
Please only emit the @update:titleEditable. If the dev don't implement the listener, then it will be disabled. Let's not have a managed inner state. It would be cleaner and allow the dev more flexibility :wink:

Reply:
What do you mean with managed inner state? The `this.editing`?

Reply:
Yes, I mean with `this.editing`.
If you just forward the event to the parent with a sync modifier `update:editing` and let the dev the option to trigger the edit from the parent.
So you'd have to change from titleEditable to `titleEditing` or something and just fire the event on click. That way if the devs want to trigger the rename, they can just change the prop and if they want to forbid it, they can just not provide the prop or not provide a sync modifier (the editing prop will be immutable)

And instead of firing $emit('start-editing') the dev can just listen to the `update:editing` event. If it's true, then the editing is on.

Reply:
I think that only works if we only allow two states for titleEditable (`true` and `false`) but not the third `"toggle"` as it is now, could that be?

Reply:
You don't need toggle then.
Always allows it, if the dev doesn't want to change it, then it can disable the editable directly.

Reply:
I removed the toggle option. When editing is allowed, the input shows after clicking.
@skjnldsv I don't know if I got your idea correctly, but I think a managed inner state with `this.titleEditing` is still necessary. Otherwise one cannot distinguish between if editing is forbidden/disabled or currently just not active.
But maybe I got it wrong. In that case feel free to adjust the code accordingly.

Reply:
No it should not be necessary.
If showing everything rely on the prop titleEditable and the dev doesn't allow it to change (so don't use the `.sync` modifier), it will not change, so clicking the title will not toggle the `titleEditable` and will have no outcome :)

If the dev want to allow the title to be editable, it should just add the sync modifier and clicking the title will then change the prop on the parent.

Reply:
Ok, I think I got it now. Thanks for the feedback @skjnldsv!